### ISSUE TITLE
IniParser.Exceptions.ParsingException exception when ini contains two ini values with [UTILITY]

### ISSUE THREAD

**Issue by TokcDK** [rickyah/ini-parser#197]
Getting exceptions when ini contains something like this:
```
[Executables]
6\binary=C:\\folder\\[UTILITY] dir1\\name.exe
7\binary=C:\\folder\\[UTILITY] dir2\\name.exe
```

Some ini cantains file/folder path in value and in file/folder name can cantain `[UTILITY]` in name but ini-parser returns parse error when trying to read the ini with `IniData INI = new FileIniDataParser().ReadFile(Path)`

```
ParsingException: Duplicate section with name 'UTILITY'. Please see configuration option Configuration.SkipInvalidLines to ignore this error. while parsing line number 81 with value '7\binary=C:\\folder\\[UTILITY] dir2\\name.exe' - IniParser version: 2.5.3.0

    IniParser.Parser.IniDataParser.ProcessSection(IniParser.Parser.StringBuffer, IniParser.Model.IniData) in IniDataParser.cs
    IniParser.Parser.IniDataParser.ProcessLine(IniParser.Parser.StringBuffer, IniParser.Model.IniData) in IniDataParser.cs
    IniParser.Parser.IniDataParser.Parse(System.IO.StringReader) in IniDataParser.cs
    IniParser.StreamIniDataParser.ReadData(System.IO.StreamReader) in StreamIniDataParser.cs
    IniParser.FileIniDataParser.ReadFile(string, System.Text.Encoding) in FileIniParser.cs
```

**Comment by rickyah** [comment]
@TokcDK, I can't seem to be able to reproduce your problem.

Maybe is something that has to do with the file encoding?
Are you using windows or linux/mac?

You can probably workaround the issue by not using `FileInidataParser` just read manually the file and parse it with `IniDataParser`

**Comment by TokcDK** [comment]
The error was while reading ModOrganizer.ini from Mod Organizer 2 where was added custom executables paths with [text] in folder name. Tests was on Win7. First time I just fixed it by edit 2.5.3 sources lineprocessing for section with somethings like `if (!line.TrimStart().StartsWith("[")) return false` and it fixed the error . But with 2.5.3 also sometimes was other problems when ini was just fully erased with one writed line from ini-parser and after this I just reverted back to more stable 2.5.2 and resolved exist problems with other ways.

Understood about `IniDataParser`.

Wil try somehow to check it again to find more precise conditions for the issue and will load ini-parser as sources to see all steps from where problem was caused.
